signac/contrib/project.py

```diff
@@ -1599,18 +1606,20 @@ def get_project(cls, root=None, search=True, **kwargs):
         """
         if root is None:
             root = os.getcwd()
-        config = load_config(root=root, local=not search)
-        if "project_dir" not in config or (
-            not search
-            and os.path.realpath(config["project_dir"]) != os.path.realpath(root)
-        ):
-            raise LookupError(
-                "Unable to determine project id for path '{}'.".format(
-                    os.path.abspath(root)
+        if not search:
+            if not _contains_config_file(root):
+                raise LookupError(
+                    f"Unable to find project at path '{os.path.abspath(root)}'."
+                )
+        else:
+            old_root = root
+            root = _locate_config_dir(root)
+            if not root:
+                raise LookupError(
+                    f"Unable to find project at path '{os.path.abspath(old_root)}'."
                 )
-            )
 
-        return cls(root=config["project_dir"], **kwargs)
+        return cls(root=root, **kwargs)
```

Comment:
Simplify this. I think this is equivalent? Maybe there's something even better in between the current state and this rewrite. My primary goal is to raise the `LookupError` only in one path.
```python
config_dir = _locate_config_dir(root) if search else root
if config_dir is None or not _contains_config_file(config_dir):
    raise LookupError(f"Unable to find project at path '{os.path.abspath(config_dir)}'.")
return cls(root=config_dir, **kwargs)
```

Reply:
The problem is that the error message becomes useless in the case that `_locate_config_dir` returns `None`. I disliked this code as well but couldn't come up with an alternative that gets around that elegantly. You could do something like this:
```
old_root = root
root = _locate_config_dir(root) if search else root
if root is not None or not os.path.isfile(_get_project_config_fn(root)):
    raise LookupError(
        f"Unable to find project at path '{os.path.abspath(old_root)}'."
    )
```
which gets around my above objection, but it still has the downside of adding an extra pair of `os` calls in the `search` case because you'll run the `os.path.isfile(_get_project_config_fn(root))` even though the current value of `root` (the output of `_locate_config_dir`) is already sufficient information. I don't think that's worthwhile. The best we can do is just set a local `err` variable in both branches and raise the `LookupError` outside the `if-else`, which sort of addresses your concern but still isn't great.

Reply:
Ultimately this is pretty minor and I don’t want to get hung up on this. If you think one of these alternatives is better, go for it.